### ISSUE TITLE
syscall stat returns success but doesn't fill the structure with data

### ISSUE THREAD

**Issue by vonasek** [mozilla/rr#2169]
STAT() on line 272 calls syscall stat which returns success but mCachedStat structure doesn't change. I guess this is some bug in rr because I never experienced this issue without rr.

```
269	bool
270	nsLocalFile::FillStatCache()
271	{
272	  if (STAT(mPath.get(), &mCachedStat) == -1) {
273	    // try lstat it may be a symlink
274	    if (LSTAT(mPath.get(), &mCachedStat) == -1) {
275	      return false;
276	    }
277	  }
278	  return true;
279	}
```

```
Thread 1 hit Breakpoint 2, nsLocalFile::FillStatCache (this=0x7fa1c55e7b80)
    at /mnt/work/opt/moz/hg-inbound-2/xpcom/io/nsLocalFileUnix.cpp:272
272	  if (STAT(mPath.get(), &mCachedStat) == -1) {

(rr) p &mCachedStat
$1 = (stat64 *) 0x7fa1c55e7ba0

(rr) p mCachedStat
$2 = {st_dev = 16493559407081481444, st_ino = 16493559407081481444, st_nlink = 16493559407081481444, st_mode = 3840206052, 
  st_uid = 3840206052, st_gid = 3840206052, __pad0 = -454761244, st_rdev = 16493559407081481444, st_size = -1953184666628070172, 
  st_blksize = -1953184666628070172, st_blocks = -1953184666628070172, st_atim = {tv_sec = -1953184666628070172, 
    tv_nsec = -1953184666628070172}, st_mtim = {tv_sec = -1953184666628070172, tv_nsec = -1953184666628070172}, st_ctim = {
    tv_sec = -1953184666628070172, tv_nsec = -1953184666628070172}, __glibc_reserved = {-1953184666628070172, -1953184666628070172, 
    -1953184666628070172}}

(rr) c
Continuing.

Thread 1 hit Breakpoint 3, __GI___xstat (vers=1, 
    name=0x23aa4a196a0c "/mnt/work/opt/moz/hg-inbound-2/_obj-browser-release-tb-fp-dbg/dist/bin/browser/chrome/browser/content/browser/ext-c-browser.js", buf=0x7fa1c55e7ba0) at ../sysdeps/unix/sysv/linux/wordsize-64/xstat.c:35
35	    return INLINE_SYSCALL (stat, 2, name, buf);

(rr) p buf
$3 = (struct stat *) 0x7fa1c55e7ba0

(rr) fin
Run till exit from #0  __GI___xstat (vers=1, 
    name=0x23aa4a196a0c "/mnt/work/opt/moz/hg-inbound-2/_obj-browser-release-tb-fp-dbg/dist/bin/browser/chrome/browser/content/browser/ext-c-browser.js", buf=0x7fa1c55e7ba0) at ../sysdeps/unix/sysv/linux/wordsize-64/xstat.c:35
0x00000000021d1418 in nsLocalFile::FillStatCache (this=0x7fa1c55e7b80) at /mnt/work/opt/moz/hg-inbound-2/xpcom/io/nsLocalFileUnix.cpp:272
272	  if (STAT(mPath.get(), &mCachedStat) == -1) {
Value returned is $4 = 0

(rr) p mCachedStat
$5 = {st_dev = 16493559407081481444, st_ino = 16493559407081481444, st_nlink = 16493559407081481444, st_mode = 3840206052, 
  st_uid = 3840206052, st_gid = 3840206052, __pad0 = -454761244, st_rdev = 16493559407081481444, st_size = -1953184666628070172, 
  st_blksize = -1953184666628070172, st_blocks = -1953184666628070172, st_atim = {tv_sec = -1953184666628070172, 
    tv_nsec = -1953184666628070172}, st_mtim = {tv_sec = -1953184666628070172, tv_nsec = -1953184666628070172}, st_ctim = {
    tv_sec = -1953184666628070172, tv_nsec = -1953184666628070172}, __glibc_reserved = {-1953184666628070172, -1953184666628070172, 
    -1953184666628070172}}
```

**Comment by rocallahan** [comment]
What version of rr? This sounds like a bug that was fixed in b8821907e0edb829fc87f9eb69ba3a088a2cfb2a (released in rr 5.1.0)

**Comment by vonasek** [comment]
You're right. It was version 5.0.0 and i can no longer reproduce the problem after updating to 5.1.0.
